src/raw/mod.rs

```diff
@@ -2817,9 +2820,11 @@ impl<T: Clone, A: Allocator + Clone> Clone for RawTable<T, A> {
 
                 // Cloning elements may fail (the clone function may panic), but the `ScopeGuard`
                 // inside the `clone_from_impl` function will take care of that, dropping all
-                // cloned elements if necessary. The `Drop` of `RawTable` takes care of the rest
-                // by freeing up the allocated memory.
-                self.clone_from_spec(source);
+                // cloned elements if necessary. Our `ScopeGuard` will clear the table.
+                self_.clone_from_spec(source);
+
+                // Disarm the scope guard if cloning was successful.
+                mem::forget(self_);
```

Comment:
I prefer `ScopeGuard::into_inner` for disarming a guard, it's more explicit.

Reply:
Done